src/SFML/Graphics/VertexBuffer.cpp

```diff
@@ -333,22 +330,15 @@ VertexBuffer::Usage VertexBuffer::getUsage() const
 ////////////////////////////////////////////////////////////
 bool VertexBuffer::isAvailable()
 {
-    std::lock_guard lock(VertexBufferImpl::isAvailableMutex);
-
-    static bool checked   = false;
-    static bool available = false;
-
-    if (!checked)
+    static const bool available = []() -> bool
     {
-        checked = true;
-
         TransientContextLock contextLock;
 
         // Make sure that extensions are initialized
         sf::priv::ensureExtensionsInit();
 
-        available = GLEXT_vertex_buffer_object;
-    }
+        return GLEXT_vertex_buffer_object;
+    }();
```

Comment:
My C++-fu isn't the best here, does the static guarantee some atomicy, or why is the mutex no longer needed?

Reply:
https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables